core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -158,15 +159,23 @@ public void registerQuery(
             Set<DynamicFilterId> lazyDynamicFilters,
             Set<DynamicFilterId> replicatedDynamicFilters)
     {
-        Map<DynamicFilterId, SettableFuture<Void>> lazyDynamicFilterFutures = lazyDynamicFilters.stream()
-                .collect(toImmutableMap(filter -> filter, filter -> SettableFuture.create()));
         dynamicFilterContexts.putIfAbsent(queryId, new DynamicFilterContext(
```

Comment:
nit: I don't think query can be registered twice. Looking at `io.trino.execution.SqlQueryExecution#start` I don't think that can happen. There is:
```
                if (!stateMachine.transitionToPlanning()) {
                    // query already started or finished
                    return;
                }

```
check.

Reply:
True, but it's unrelated to this PR. It should be tackled separately, if needed.